src/avlizer_confluent.erl

```diff
@@ -17,29 +17,68 @@
-%%      Caller is also given liberty to assign or generate fingerprint for
-%%      the schema before registeration. see `register_schema_with_fp/3'
+%% @doc This module encodes and decodes Avro binary data tagged with Confluent
+%% Schema Registry integer registry IDs or CRC64 fingerprints.
 %%
-%%  Registration ID is more 'compact' in payload (only 5 bytes overhead).
-%%  Fingerprint is more static (can be obtained at compile time) comparing
-%%  to the assigned registration ID which is assigned at runtime, and
-%%  most likely different across registries.
+%% See https://www.confluent.io/confluent-schema-registry
+%%
+%% It looks up schemas in the Schema Registry using its HTTP API.
+%% See https://docs.confluent.io/current/schema-registry/develop/api.html
+%%
+%% It maintains a cache, allowing encoders and decoders to be looked
+%% up using Schema Registry IDs or name + fingerprint.
+%%
+%% It implements two data serializers:
+%%
+%%   1. Confluent serialization wire format
+%%
+%%      https://docs.confluent.io/
+%%          current/schema-registry/serializer-formatter.html#wire-format
+%%
+%%      Avro binary encoded objects are prefixed with a five-byte tag.
+%%      The first byte indicates the Confluent serialization format
+%%      version number, currently always 0.
+%%      The following four bytes encode the integer schema ID as
+%%      returned from the Schema Registry in network byte order.
+%%
+%%   2. Fingerprint prefixed data
+%%
+%%      When used without a schema registry, it's common to prefix
+%%      binary data with a hash of the schema that created it.
+%%
+%%      See https://avro.apache.org/docs/1.8.2/spec.html#schema_fingerprints
+%%      and https://avro.apache.org/docs/1.8.2/spec.html#single_object_encoding_spec
+%%
+%%      In the Avro "Single-object encoding", the Avro binary data is prefixed
+%%      with a two-byte marker, C3 01, to show that the message is Avro and
+%%      uses this single-record format (version 1). That is followed by the
+%%      the 8-byte little-endian CRC-64-AVRO fingerprint of the
+%%      object's schema.
+%%
+%%      The CRC64 hash is uncommon, but used because it is shorter
+%%      than e.g. MD5, while still being good enough to detect
+%%      collisions. The CRC64 hash is implemented in
+%%      `erlavro:crc64_fingerprint/1'.
+%%
+%%      This module's cache supports using the name and fingerprint as
+%%      a key. The name is normally the Avro "full name"
+%%      (https://avro.apache.org/docs/1.8.2/spec.html#names), e.g.
+%%      com.example.X. The fingerprint is CRC64 by default. You can
+%%      also register a name with your own fingerprint using
+%%      `register_schema_with_fp/3'.
+%%
+%%  The Schema Registry ID is more compact in the payload (only five
+%%  bytes overhead), but relies on the registry.
+%%
+%%  The fingerprint is more static, and can be obtained at compile
+%%  time, but it is harder to share with other applications and evolve
```

Comment:
To share schema by fp, it is not harder than registration ID. Since they are essentially just two numbers.
One may argue that without fp as a part of the encoded payload makes it harder to share (maybe that’s what you meant)
But I would phrase it as more freedom on choosing means to share I’d.

Reply:
E.g fp as payload meta data (it’s common to have envelops )
Or fp in kafka message header. Etc.
Even we in some cases did double registration: the id returned from fp registration can also be used to share the schema.